pkg/serviceaccount/util.go

```diff
@@ -25,11 +25,35 @@ import (
 
 // UserInfo returns a user.Info interface for the given namespace, service account name and UID
 func UserInfo(namespace, name, uid string) user.Info {
-	return &user.DefaultInfo{
-		Name:   apiserverserviceaccount.MakeUsername(namespace, name),
-		UID:    uid,
-		Groups: apiserverserviceaccount.MakeGroupNames(namespace),
+	return (&ServiceAccountInfo{
+		Name:      name,
+		Namespace: namespace,
+		UID:       uid,
+	}).UserInfo()
+}
+
+type ServiceAccountInfo struct {
+	Name, Namespace, UID string
+	PodName, PodUID      string
+}
+
+func (sa *ServiceAccountInfo) UserInfo() user.Info {
+	const (
+		podNameKey = "kubernetes.io/pod.name"
+		podUIDKey  = "kubernetes.io/pod.uid"
```

Comment:
@kubernetes/sig-auth-api-reviews PTAL. These will be the keys of the pod uid and name embedded in ExtraInfo.